src/cpp/rtps/transport/TCPChannelResource.cpp

```diff
@@ -375,6 +375,9 @@ void TCPChannelResource::set_socket_options(
         asio::basic_socket<asio::ip::tcp>& socket,
         const TCPTransportDescriptor* options)
 {
+    // Options setting should be done before connection is established
```

Comment:
Please double check this is indeed the case. At least today it is I believe, and if willing to change that in the future, it should be assured the socket is modified in a thread safe manner.